lib/build.py

```diff
@@ -181,137 +157,307 @@ def build_arg_get(arg):
       ml.visit_topdown(tree)
    else:
       ml.visit(tree)
+   if (ml.instruction_total_ct > 0):
+      if (ml.miss_ct == 0):
+         ml.inst_prev.checkout()
+      ml.inst_prev.ready()
 
    # Check that all build arguments were consumed.
    if (len(cli.build_arg) != 0):
       ch.FATAL("--build-arg: not consumed: " + " ".join(cli.build_arg.keys()))
 
    # Print summary & we're done.
-   if (ml.instruction_ct == 0):
+   if (ml.instruction_total_ct == 0):
       ch.FATAL("no instructions found: %s" % cli.file)
-   assert (image_i + 1 == image_ct)  # should have errored already if not
-   if (cli.force):
+   assert (ml.inst_prev.image_i + 1 == image_ct)  # should’ve errored already
+   if (cli.force and ml.miss_ct != 0):
       if (fakeroot_config.inject_ct == 0):
          assert (not fakeroot_config.init_done)
          ch.WARNING("--force specified, but nothing to do")
       else:
          ch.INFO("--force: init OK & modified %d RUN instructions"
                  % fakeroot_config.inject_ct)
    ch.INFO("grown in %d instructions: %s"
-           % (ml.instruction_ct, images[image_i]))
+           % (ml.instruction_total_ct, ml.inst_prev.image))
+   # FIXME: remove when we're done encouraging people to use the build cache.
+   if (isinstance(bu.cache, bu.Disabled_Cache)):
+      ch.INFO("build slow? consider enabling the new build cache",
+              "https://hpc.github.io/charliecloud/command-usage.html#build-cache")
```

Comment:
The command reference for the build-cache subcommand isn't very useful. It doesn't mention how to enable it nor the dependencies required. Perhaps a link to ch-image, a link to an added faq section, or a different message.

Reply:
It should go to the “Build cache” section; if not that’s an error.